### PR TITLE
feat: fetch announcements

### PR DESCRIPTION
**Pull request by marcomariscal**
fetches announcements based on supplied filters.

next step (after merging) will be to tie this together with `checkStealthAddress`

### CODE REVIEW

File: src/lib/createStealthClient.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { createPublicClient, http, type PublicClient } from 'viem';
+import type {
+  ClientParams,
+  InitializedStealthActions,
```
Comment (garyghayrat) [inline src/lib/createStealthClient.ts:4]: Unused import

File: examples/getAnnouncements/index.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import {
+  ERC5564_CONTRACT,
+  VALID_SCHEME_ID,
+  createStealthClient,
+  getAnnouncements,
+} from 'stealth-address-sdk';
+
+// Example parameters
+const chainId = 11155111; // Example chain ID for Sepolia
+const rpcUrl = 'https://sepolia.infura.io/v3/YOUR_INFURA_PROJECT_ID';
```
Comment (garyghayrat) [inline examples/getAnnouncements/index.ts:10]: Let's include a .env.example file for the RPC url.

File: src/lib/helpers/test/setupTestEnv.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { createStealthClient } from '../..';
+import { ERC5564_CONTRACT } from '../../..';
+import type { VALID_CHAIN_IDS } from '../types';
+
+function setupTestEnv() {
+  if (!process.env.TEST_CHAIN_ID) {
```
Comment (garyghayrat) [inline src/lib/helpers/test/setupTestEnv.ts:6]: Let's create `.env.example` files for the env vars in this file.

File: src/lib/actions/getAnnouncements/getAnnouncements.test.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    const announcements = await stealthClient.getAnnouncements({
+      ERC5564Address,
+      args: {
+        stealthAddress,
+      },
+      fromBlock,
+    });
+
+    expect(announcements[0].stealthAddress).toBe(stealthAddress);
+  });
```
Comment (garyghayrat) [inline src/lib/actions/getAnnouncements/getAnnouncements.test.ts:61]: Great start, prob will want to add more tests for multiple announcements and to check other filters also work properly.

File: src/lib/createStealthClient.ts (hunk 1 above)
Comment (marcomariscal) [reply src/lib/createStealthClient.ts]: updated

File: examples/getAnnouncements/index.ts (hunk 2 above)
Comment (marcomariscal) [reply examples/getAnnouncements/index.ts:10]: good call

File: src/lib/actions/getAnnouncements/getAnnouncements.test.ts (hunk 4 above)
Comment (marcomariscal) [reply src/lib/actions/getAnnouncements/getAnnouncements.test.ts:60]: yesser! added a few more tests for the other filters

File: src/lib/createStealthClient.ts (hunk 1 above)
Comment (garyghayrat) [reply src/lib/createStealthClient.ts]: Commit: https://github.com/ScopeLift/stealth-address-sdk/pull/15/commits/369721869ddf82d53cc5f0f82352e290663cfb24

File: examples/getAnnouncements/index.ts (hunk 2 above)
Comment (garyghayrat) [reply examples/getAnnouncements/index.ts:10]: [a5b7541](https://github.com/ScopeLift/stealth-address-sdk/pull/15/commits/a5b7541b0be7de96edf9735392166356ce09eeda)

File: src/lib/helpers/test/setupTestEnv.ts (hunk 3 above)
Comment (garyghayrat) [reply src/lib/helpers/test/setupTestEnv.ts:6]: [a5b7541](https://github.com/ScopeLift/stealth-address-sdk/pull/15/commits/a5b7541b0be7de96edf9735392166356ce09eeda)

File: src/lib/actions/getAnnouncements/getAnnouncements.test.ts (hunk 4 above)
Comment (garyghayrat) [reply src/lib/actions/getAnnouncements/getAnnouncements.test.ts:60]: [02acbee](https://github.com/ScopeLift/stealth-address-sdk/pull/15/commits/02acbeecf704bf94a6c7bde414775379eaf4d1c7)

File: examples/getAnnouncements/index.ts (hunk 2 above)
Comment (marcomariscal) [reply examples/getAnnouncements/index.ts]: updated!

Comment (garyghayrat) [reply examples/getAnnouncements/index.ts]: [fd49c40](https://github.com/ScopeLift/stealth-address-sdk/pull/15/commits/fd49c40a2eb2e029e577c0b7204f6044deedd7df)

File: src/lib/actions/getAnnouncements/getAnnouncements.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+      ephemeralPubKey: args.ephemeralPubKey,
+      metadata: args.metadata,
+      ...log,
+    };
+  });
+
+  return announcements;
+}
+
+const fetchLogsInChunks = async ({
```
Comment (garyghayrat) [inline src/lib/actions/getAnnouncements/getAnnouncements.ts:50]: Needs documentation here also

**Review by garyghayrat — APPROVED** [review]
Could use more docs explaining things but otherwise lgtm!

File: src/lib/actions/getAnnouncements/getAnnouncements.ts (hunk 5 above)
Comment (marcomariscal) [reply src/lib/actions/getAnnouncements/getAnnouncements.ts:72]: yee updated!

**Pull request merged**